data_store/db/entities.py

```diff
@@ -416,7 +416,6 @@ class ProgrammeJunction(BaseModel):
     )
     programme_id: Mapped[GUID] = sqla.orm.mapped_column(sqla.ForeignKey("programme_dim.id"), nullable=False)
     reporting_round_id: Mapped[GUID] = sqla.orm.mapped_column(sqla.ForeignKey("reporting_round.id"), nullable=False)
-    reporting_round = sqla.Column(sqla.Integer, nullable=True)
```

Comment:
Wouldn't there be a migration with this?

Reply:
Found it!

Reply:
Would've kept the to PR's one.

Reply:
it's intentional to not do them together, because doing them together is 'theoretically' broken (due to how sqlalchemy builds SQL queries). happy to chat through if useful

Reply:
actually I'll just explain here.

SQLAlchemy doesn't do a `select *` statement - it will do an explicit `select col1, col2, col` for all columns it knows about based on the python model.

Because we run migrations before deploying the code (as is required), if you remove the python column definitions in the same PR a you have a migration to remove the columns, you can end up with the old code trying to query columns that have been dropped by the migration before the new code has deployed.